sdk/resourcemanager/Azure.ResourceManager.Core/src/Operation/PhArmOperation.cs

```diff
@@ -0,0 +1,139 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// A calss representing an arm operation wrapper object.
+    /// </summary>
+    /// <typeparam name="TOperations"> The <see cref="ResourceOperationsBase"/> to convert TModel into. </typeparam>
+    /// <typeparam name="TModel"> The model returned by existing Operation methods. </typeparam>
+    public class PhArmOperation<TOperations, TModel> : ArmOperation<TOperations>
+        where TOperations : class
+        where TModel : class
+    {
+        private readonly Func<TModel, TOperations> _converter;
+        private readonly Operation<TModel> _wrappedOperation;
+        private readonly ArmOperation<TModel> _wrappedResponseOperation;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PhArmOperation{TOperations, TModel}"/> class for mocking.
+        /// </summary>
+        protected PhArmOperation()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PhArmOperation{TOperations, TModel}"/> class.
+        /// </summary>
+        /// <param name="wrapped"> The results to wrap. </param>
+        /// <param name="converter"> The function used to convert from existing type to new type. </param>
+        public PhArmOperation(Operation<TModel> wrapped, Func<TModel, TOperations> converter)
+        {
+            if (wrapped is null)
+                throw new ArgumentNullException(nameof(wrapped));
+
+            if (converter is null)
+                throw new ArgumentNullException(nameof(converter));
+
+            _wrappedOperation = wrapped;
+            _converter = converter;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PhArmOperation{TOperations, TModel}"/> class.
+        /// </summary>
+        /// <param name="wrapped"> The results to wrap. </param>
+        /// <param name="converter"> The function used to convert from existing type to new type. </param>
+        public PhArmOperation(Response<TModel> wrapped, Func<TModel, TOperations> converter)
+        {
+            if (wrapped is null)
+                throw new ArgumentNullException(nameof(wrapped));
+
+            if (converter is null)
+                throw new ArgumentNullException(nameof(converter));
+
+            _wrappedResponseOperation = new PhValueArmOperation<TModel>(wrapped);
+            _converter = converter;
+        }
+
+        private bool _doesWrapOperation => _wrappedResponseOperation is null;
+
+        /// <inheritdoc/>
+        public override string Id => _wrappedOperation?.Id;
+
+        /// <inheritdoc/>
+        public override TOperations Value => _converter(_doesWrapOperation ? _wrappedOperation.Value : _wrappedResponseOperation.Value);
```

Comment:
```suggestion
        public override TOperations Value => _converter(_doesWrapOperation ? _wrappedOperation?.Value : _wrappedResponseOperation?.Value);
```

Reply:
the if check on _doesWrapOperation should guarantee my nullability